homeassistant/components/xiaomi_miio/alarm_control_panel.py

```diff
@@ -76,6 +77,29 @@ async def _try_command(self, mask_error, func, *args, **kwargs):
         except DeviceException as exc:
             _LOGGER.error(mask_error, exc)
 
+    @callback
+    def alarm_callback(self, action, params):
+        """Push from gateway."""
+        _LOGGER.debug("Got new alarm_callback: %s", action)
+        if action == "alarm_triggering":
+            self._attr_state = STATE_ALARM_TRIGGERED
+            self.schedule_update_ha_state()
```

Comment:
Will this run in the event loop or in another thread?

Reply:
It is a async callback from the running event loop, so yes I think this runs inside the event loop.
Is that okay?